### PR TITLE
Add new modules documentation

### PR DESCRIPTION
**Pull request by jfy133**
Transferred from the nf-core/modules README

### CODE REVIEW

File: markdown/developers/adding_modules.md [hunk 1]
```diff
@@ -0,0 +1,406 @@
+---
+title: Adding a new module
+subtitle: Follow this walkthrough for adding and reviewing new nf-core modules
+---
+
+## Adding a new module file
+
+If you decide to upload a module to `nf-core/modules` then this will ensure that it will become available to all nf-core pipelines, and to everyone within the Nextflow community! See [`modules/`](modules)
```
Comment (ewels) [inline markdown/developers/adding_modules.md:8]: Link is now wrong I think..

File: markdown/developers/adding_modules.md [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+for examples.
+
+## Terminology
+
+The features offered by Nextflow DSL2 can be used in various ways depending on the granularity with which you would like to write pipelines. Please see the listing below for the hierarchy and associated terminology we have decided to use when referring to DSL2 components:
+
+- *Module*: A `process` that can be used within different pipelines and is as atomic as possible i.e. cannot be split into another module. An example of this would be a module file containing the process definition for a single tool such as `FastQC`. At present, this repository has been created to only host atomic module files that should be added to the [`modules/`](modules/) directory along with the required documentation and tests.
```
Comment (ewels) [inline markdown/developers/adding_modules.md:15]: Could we make these have their own level 3 headings please? Then we can link to them directly.

File: markdown/developers/adding_modules.md [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+- Add yourself to the `Assignees` so we can track who is working on the module
+
+### nf-core modules create
+
+We have implemented a number of commands in the `nf-core/tools` package to make it incredibly easy for you to create and contribute your own modules to nf-core/modules.
+
+1. Install the latest version of [`nf-core/tools`](https://github.com/nf-core/tools#installation) (`>=2.0`)
+2. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=21.04.0`)
+3. Install any of [`Docker`](https://docs.docker.com/engine/installation/), [`Singularity`](https://www.sylabs.io/guides/3.0/user-guide/) or [`Conda`](https://conda.io/miniconda.html)
+4. [Fork and clone this repo locally](#uploading-to-nf-coremodules)
```
Comment (ewels) [inline markdown/developers/adding_modules.md:42]: Needs rewriting

**Review by ewels — CHANGES_REQUESTED** [review]
Looking good! Few text tweaks needed. I started going through making suggested changes for lots of heading levels (number of `#`s) as it's much more important for the website than it is on GitHub. But then I changed my mind and thought I should see how it renders first. But they may need some attention.

File: markdown/developers/adding_modules.md (hunk 3 above)
Comment (jfy133) [reply markdown/developers/adding_modules.md:47]: Nore sure what you mean here?

**Comment by jfy133** [comment]
I've just updated the headers anyway because I shuffled a few sections around, and I replaced all references to repository directories with URLS (I think I got them all, but please let me know if you identify more)

I also added an additional step in the workflow to check guidelines before linting.

File: markdown/developers/adding_modules.md (hunk 3 above)
Comment (ewels) [reply markdown/developers/adding_modules.md:47]: "This repo" makes sense when in a readme of the modules repo. Not when in standalone docs in the website though.

File: markdown/developers/adding_modules.md [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+
+### New module workflow
+
+We have implemented a number of commands in the `nf-core/tools` package to make it incredibly easy for you to create and contribute your own modules to nf-core/modules.
+
+1. Install the latest version of [`nf-core/tools`](https://github.com/nf-core/tools#installation) (`>=2.1`)
+2. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=21.04.0`)
+3. Install any of [`Docker`](https://docs.docker.com/engine/installation/), [`Singularity`](https://www.sylabs.io/guides/3.0/user-guide/) or [`Conda`](https://conda.io/miniconda.html)
+4. [Fork and clone this repo locally](#uploading-to-nf-coremodules)
+5. Set up git by adding a new remote of the nf-core git repo called `upstream`
```
Comment (ewels) [inline markdown/developers/adding_modules.md:48]:
```suggestion
5. Set up git on your computer by adding a new git remote of the main nf-core repo called `upstream`
```

**Review by ewels — APPROVED** [review]
LGTM on my phone! Good improvements ✅ Just the reference to "this repo" remaining that I can see from a quick skim read.

**Comment by jfy133** [comment]
@drpatelh any thoughts before I merge?

**Review by drpatelh — APPROVED** [review]
LGTM! These should evolve over time anyway but defo a good start! Thanks @jfy133 ! Apologies for the delay.

**Pull request merged**